##### heading direction
# LEFT 0
# STRAIGHT 1
# RIGHT 2

##### speed
# SLOW DOWN 0
# SPEED UP 1
# MAINTAIN 2
# STOP 3

string head_dir
string speed